firstvoices/backend/tests/test_apis/test_dictionary_cleanup_api.py

```diff
@@ -2,64 +2,94 @@
 from unittest.mock import ANY
 
 import pytest
-from django.urls import reverse
 
+from backend.models import DictionaryEntry
 from backend.models.constants import AppRole, Visibility
 from backend.tests import factories
 from backend.tests.test_apis.base_api_test import BaseApiTest
 
+RECALCULATION_CONFIRMATION_MESSAGE = {"message": "Recalculation has been queued."}
+RECALCULATION_PREVIEW_CONFIRMATION_MESSAGE = {
+    "message": "Recalculation preview has been queued."
+}
 
-class TestDictionaryCleanupPreview(BaseApiTest):
+
+class TestDictionaryCleanup(BaseApiTest):
     API_DETAIL_VIEW = "api:site-detail"
 
-    def get_detail_endpoint(self, key):
-        return (
-            reverse(self.API_DETAIL_VIEW, current_app=self.APP_NAME, args=[key])
-            + "dictionary-cleanup/preview/"
-        )
+    def get_cleanup_endpoint(self, key, is_preview):
+        if is_preview:
+            return self.get_detail_endpoint(key) + "dictionary-cleanup/preview/"
+        else:
+            return self.get_detail_endpoint(key) + "dictionary-cleanup/"
 
     @pytest.mark.django_db
-    def test_recalculate_preview_get_404(self):
+    @pytest.mark.parametrize("is_preview", [True, False])
```

Comment:
Rather than having all these tests check two different endpoints based on a flag, let's have two different subclasses with different settings and shared logic. Same amount of reuse, but it's much easier to understand and reason about when each class has one job instead of two